sp-sim/src/gimlet.rs

```diff
@@ -364,6 +400,34 @@ impl UdpTask {
     }
 }
 
+async fn handle_request<'a>(
+    gateway_addresses: &Mutex<[Option<SocketAddr>; 2]>,
+    handler: &mut Handler,
+    recv: Result<(&[u8], SocketAddr)>,
+    server: &'a mut SpServer,
+    responsiveness: Responsiveness,
+    port_num: usize,
```

Comment:
I mentioned this in another review, but I think this should be `assert!(port_num == 1 || portnum == 2)`

Reply:
Yeah, good call. There's one spot where I end up converting this into an enum with two cases; I think I should just do that for all of them and ditch `usize` entirely.